icu-date.lua

```diff
@@ -14,6 +14,7 @@ local _M = {}
 _M.__index = _M
 
 local function call_fn(name, ...)
+  assert(select(1, ...))
```

Comment:
1. `select(1, ...)` is equal to `...` (selects all values starting from 1st). If you intend to fix it in that way () should be used here (and a message should be added): https://github.com/tarantool/tarantool/blob/2c351664bcd3f923cfb4c4892cb33db528c05e32/src/box/lua/tuple.lua#L304

2. I cannot understand why this abstract function should check 1st argument for being non-null. I think functions in public API (such as `parse`) should check arguments properly (including self). Yep, it is a bit more work, but good error messages are worth it.

Reply:
Ok. I've added more informative error messages and now public functions check `self` type